tfjs-layers/src/engine/base_random_layer.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright 2022 CodeSmith LLC
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ * =============================================================================
+ */
+
+import { LayerArgs, Layer } from './topology';
+import { randomGamma, randomNormal} from '@tensorflow/tfjs-core';
+import { randomStandardNormal, randomUniform } from '@tensorflow/tfjs-core';
+import { ValueError } from '../errors';
+
+export declare interface BaseRandomLayerArgs extends LayerArgs {}
+
+export type RNGTypes = {
+  [key: string]: Function;
+};
+
+export abstract class BaseRandomLayer extends Layer {
+  // A layer handle the random number creation and savemodel behavior.
+  /** @nocollapse */
+  static className = 'RandomWidth';
+  randomGenerator: Function;
+  private rngType: string;
```

Comment:
This can reference RNGTypes to be more specific (I know they're both the `string` type right now, but that may change if the other change I mentioned can be applied).
```ts
private rngType: keyof RNGTypes
```